### ISSUE TITLE
`docCommentsBeforeModifiers` acting weird with enum cases named as modifiers

### ISSUE THREAD

**Issue by erenbesel** [nicklockwood/SwiftFormat#1923]
The new rule `docCommentsBeforeModifiers` behaves unexpectedly if there are enum cases named as modifiers such as `required`, `optional` where it moves the next doc comment between the `case` and the `caseName`.

Example:
```
enum Policy: String {
   
     /// Required policy
     case required

    /// Optional policy
    case optional

    /// Other policy
    case other
}
```

The rules changes this to:
```
enum Policy: String {
   
    /// Required policy
    case /// Optional policy
    required

    case /// Other policy
    optional

    case other
}
```

It seems to treat `case "modifier"` part as a function name and puts the next doc comment in between.

We caught this and disabled the rule for now, but could be problematic for non aware users.

I also wrote a test in the `DocCommentsBeforeModifiersTests` file and confirmed the same behavior. Let me know if you need more info!

Update: Seems this happens when there are both `required` AND `optional` cases together in the enum. It works correctly if there is only one of them!

**Comment by asos-michaelwaterfall** [comment]
We're seeing this too after upgrading from 0.54.6 to 0.55.0

**Comment by mwaterfall** [comment]
Here's two tests that fail (a modified `testCaseCommentsNotMangled` and a new test with a function called `dynamic`):

```swift
func testCaseCommentsNotMangled() {
    let input = """
    enum Symbol: CustomStringConvertible, Hashable {
        /// A named variable
        case variable(String)

        /// An infix operator
        case infix(String)

        /// A prefix operator
        case prefix(String)

        /// A postfix operator
        case postfix

        /// Required
        case required

        /// Optional
        case optional

        /// Open
        case open

        /// Other
        case other
    }
    """

    testFormatting(for: input, rule: .docCommentsBeforeModifiers)
}

func testDynamicFunctionName() {
    let input = """
    /// Tint color
    static let tintColor = UIColor.dynamic(light: .fullBlack, dark: .white)
    /// Text color
    static let textColor = UIColor.dynamic(light: .fullBlack, dark: .white)
    """

    testFormatting(for: input, rule: .docCommentsBeforeModifiers)
}
```

**Comment by nicklockwood** [comment]
@erenbesel fixed in 0.55.1
